presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -282,7 +282,8 @@ else if (c == ')') {
         verify(parameters.size() == fieldNames.size() || fieldNames.isEmpty(), "Number of parameters and fieldNames for ROW type doesn't match");
         List<TypeSignatureParameter> result = new ArrayList<>();
         for (int i = 0; i < parameters.size(); i++) {
-            String fieldName = fieldNames.isEmpty() ? format("field%d", i) : fieldNames.get(i);
+            // Use "field" + i instead of String.format to avoid expensive string formatting
```

Comment:
Add this comment just in case someone change this back to `format(..)` in the future.